xds/internal/server/conn_wrapper.go

```diff
@@ -121,6 +132,26 @@ func (c *connWrapper) XDSHandshakeInfo() (*xdsinternal.HandshakeInfo, error) {
 	return xdsinternal.NewHandshakeInfo(c.rootProvider, c.identityProvider, nil, secCfg.RequireClientCert), nil
 }
 
+func (c *connWrapper) Callback(st transport.ServerTransport) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.draining {
+		st.Drain("draining")
```

Comment:
What does this call to `Drain` do and why is this needed?

Reply:
In the case where you get a conn accepted, give up the mu, and need to drain eventually. It needs to wait for the server transport to be created. This was Doug's ingenious solution :).